### ISSUE TITLE
`ref` parameters to `ref struct` types should be `scoped ref` by default

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#62691]
`ref` parameters to `ref struct` types should be `scoped ref` by default.

No errors should be reported for the following:
```csharp
ref struct R
{
    public ref int F;
    public R(ref int i) { F = ref i; }
}

class Program
{
    static ref R F()
    {
        int i = 42;
        var r = new R(ref i);
        return ref ReturnRef(ref r); // ok
    }
    
    static ref R ReturnRef(ref R r) => throw null;
}
```

**Comment by cston** [comment]
cc @jaredpar

**Comment by jaredpar** [comment]
Corresponding spec change https://github.com/dotnet/csharplang/pull/6285
